src/main/java/cubicchunks/CustomCubicChunksWorldType.java

```diff
@@ -44,10 +45,10 @@ public CustomCubicChunksWorldType() {
 		ServerCubeCache cubeCache = world.getCubeCache();
 		
 		// init the worldgen pipeline
-		GeneratorStage terrain = new GeneratorStage("terrain");
-		GeneratorStage surface = new GeneratorStage("surface");
-		GeneratorStage features = new GeneratorStage("features");
-		GeneratorStage population = new GeneratorStage("population");		
+		GeneratorStage terrain = new IndependentGeneratorStage("terrain");
```

Comment:
For now I decided not to go through and update the existing terrain generator code. I didn't have any time to go through all of the classes involved to grasp the whole picture.